core/src/main/java/io/grpc/internal/ServerImpl.java

```diff
@@ -278,29 +281,8 @@ public ServerStreamListener streamCreated(final ServerStream stream, final Strin
       // Run in serializingExecutor so jumpListener.setListener() is called before any callbacks
       // are delivered, including any errors. Callbacks can still be triggered, but they will be
       // queued.
-      serializingExecutor.execute(new Runnable() {
-          @Override
-          public void run() {
-            ServerStreamListener listener = NOOP_LISTENER;
-            try {
-              ServerMethodDefinition<?, ?> method = registry.lookupMethod(methodName);
-              if (method == null) {
-                stream.close(
-                    Status.UNIMPLEMENTED.withDescription("Method not found: " + methodName),
-                    new Metadata());
-                timeout.cancel(true);
-                return;
-              }
-              listener = startCall(stream, methodName, method, timeout, headers);
-            } catch (Throwable t) {
-              stream.close(Status.fromThrowable(t), new Metadata());
-              timeout.cancel(true);
-              throw Throwables.propagate(t);
-            } finally {
-              jumpListener.setListener(listener);
```

Comment:
I would _really_ like the `setListener()` to remain in this same `streamCreated()` method, so it is easy to learn when it happens and the hackery involved.

I'm okay with moving some of the contents out here, though. How about something that looks more like:
```java
serializingExecutor.execute(new Runnable() {
    @Override
    public void run() {
      ServerStreamListener listener = NOOP_LISTENER;
      try {
        listener = startCall(...);
      } finally {
        jumpListener.setListener(listener);
      }
    }
});
```

And `startCall` could return `NOOP_LISTENER`. (And it's okay if startCall is named something else.)